### ISSUE TITLE
IPv6 CIDR style zones can result in incorrect reverse zones

### ISSUE THREAD

**Issue by chrisohaver** [coredns/coredns#4656]
**What happened**:

In CoreDNS 1.8.4, IPv6 CIDR style zones can expand to an incorrect set of reverse zones.

e.g.

`2001:db8::/115` Results in 2 `/120`'s...

```

/120  /116  /112      /96       /80       /64       /48       /32       /16
0.    0.    0.0.0.0.  0.0.0.0.  0.0.0.0.  0.0.0.0.  0.0.0.0.  8.b.d.0.  1.0.0.2.  ip6.arpa.
1.    0.    0.0.0.0.  0.0.0.0.  0.0.0.0.  0.0.0.0.  0.0.0.0.  8.b.d.0.  1.0.0.2.  ip6.arpa.
```

**What you expected to happen**:

`2001:db8::/115` Expected result is 2 `/116`'s ...

```
/116  /112      /96       /80       /64       /48       /32       /16
0.    0.0.0.0.  0.0.0.0.  0.0.0.0.  0.0.0.0.  0.0.0.0.  8.b.d.0.  1.0.0.2.  ip6.arpa.
1.    0.0.0.0.  0.0.0.0.  0.0.0.0.  0.0.0.0.  0.0.0.0.  8.b.d.0.  1.0.0.2.  ip6.arpa.```

```

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

**Environment**:

- the version of CoreDNS:  1.8.4
- Corefile:
- logs, if applicable:
- OS (e.g: `cat /etc/os-release`):
- Others:

**Comment by miekg** [comment]
My idea of using go-cidr was to avoid things like this. Maybe we should a)
fork b) see how responsive upstream is.

**Comment by chrisohaver** [comment]
> My idea of using go-cidr was to avoid things like this. Maybe we should a)
fork b) see how responsive upstream is.

The bug is in our code though. Fixed in #4652.
